node/container.go

```diff
@@ -136,13 +137,13 @@ func (a *HostAgent) StartContainer(cancel <-chan interface{}, serviceID string,
 	// Update the service with the complete image name
 	evaluatedService.ImageID = imageName
 
-       // Establish a connection to the master
-       masterClient, err := master.NewClient(a.master)
-       if err != nil {
-               logger.WithField("master", a.master).WithError(err).Debug("Could not connect to the master")
-               return nil, nil, err
-       }
-       defer masterClient.Close()
+	// Establish a connection to the master
```

Comment:
If I'm not mistaken, I think you can remove lines 141-149, and remove all references to `systemUser`. In other words, if `systemUser` is no longer injected into the container, I believe that you will find it's not used anywhere else in this file. Removing the call to  `masterClient.GetSystemUser()` will also reduce the number of RPC calls back to the master for each service started on a delegate, which should help performance when the master is trying to start >1000 services